src/views/sandwich-view.tsx

```diff
@@ -18,7 +18,53 @@ interface SandwichViewProps {
   glCanvas: HTMLCanvasElement
 }
 
-class SandwichView extends StatelessComponent<SandwichViewProps> {
+interface SandwichViewState {
+  isDragging: boolean
+  resizerPos: number | null
+  initialPos: number
+  delta: number
+}
+
+class SandwichView extends Component<SandwichViewProps, SandwichViewState> {
+  constructor(props: SandwichViewProps) {
+    super(props)
+    this.state = {
+      isDragging: false,
+      resizerPos: null,
+      initialPos: 0,
+      delta: 0,
```

Comment:
Rather than having multiple nullable properties & booleans expressing a single idea, I like to group together this information into a single concept where none of the properties are nullable. You can see that in the diff I attached to the top-level review comment:

```diff
+interface ActiveDrag {
+  lastMouseX: number
+}
+
 interface SandwichViewState {
-  isDragging: boolean
-  resizerPos: number | null
-  initialPos: number
-  delta: number
+  activeDrag: ActiveDrag | null
 }
```

This is an alternative structure to e.g. doing:

```
interface SandwichViewState {
  lastMouseX: number | null
  isDragging: boolean
}
```

and I think it's preferable, because then you don't have to consider the case of `isDragging` being `true` while `lastMouseX` is `null`.